File: java/javadpbo/src/com/sem4dpbo/week3/praktikum3/latihanpraktikum3_java/MainDevice.java
package com.sem4dpbo.week3.praktikum3.latihanpraktikum3_java;

public class MainDevice {
    public static void main(String[] args) {
        // class processor
        Processor dataProc = new Processor("Intel", 3200000);

        //class disk
        Disk dataDisk = new Disk("SSD", "256 GB", 800000L);

        // class ram
        Ram dataRam = new Ram("16 GB", 1200000L);

        // class pc
        PC dataPc = new PC(dataProc, dataDisk, dataRam);

        // menampilkan hasil akhir
        dataPc.countTotal();
        dataPc.printPc();
    }
}

File: cpp/3rd_Week/Praktikum/latihanpraktikum3_cpp/MainDevice.cpp
#include <bits/stdc++.h>
#include <string>

using namespace std;

// #include "Processor.cpp"
// #include "Disk.cpp"
// #include "Ram.cpp"
#include "Pc.cpp"

int main()
{
    // class processor
    Processor dataProc("Intel", 3200000);
    
    //class disk
    Disk dataDisk("SSD", "256 GB", 800000);

    // class ram
    Ram dataRam("16 GB", 1200000);

    // class pc
    Pc dataPc(dataProc, dataDisk, dataRam);

    // menampilkan hasil akhir
    dataPc.countTotal();
    dataPc.printPc();

    return 0;
}
